src/jshint.js

```diff
@@ -210,6 +210,10 @@ var JSHINT = (function() {
       combine(predefined, vars.ecmaIdentifiers[6]);
     }
 
+    if (state.option.strict === "global" && "globalstrict" in state.option) {
```

Comment:
I'd add a note explaining _why_ you've used `prop in obj`, ie. that both `false` and `true` must be considered.